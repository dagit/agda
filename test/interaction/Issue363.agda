module Issue363 where

infixl 0 _>>=_

postulate
  A     : Set
  x     : A
  _>>=_ : A → (A → A) → A
  P     : A → Set

lemma : P ((x >>= λ x → x) >>= λ x → x)
lemma = {!!}

-- The type of the goal above was printed as follows:
--
--   P (x >>= λ x' → x' >>= λ x' → x')
--
-- This is not correct.
