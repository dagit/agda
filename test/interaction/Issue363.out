agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-status-action "")
agda2_mode_code (agda2-info-action "*All Goals*" "?0 : P (x >>= (λ x' → x') >>= (λ x' → x'))
")
agda2_mode_code (agda2-goals-action '(0))
